go/adbc/driver/bigquery/connection.go

```diff
@@ -617,7 +617,32 @@ func (c *connectionImpl) getTableSchemaWithFilter(ctx context.Context, catalog *
 		metadata["MaterializedView.Query"] = md.MaterializedView.Query
 		metadata["MaterializedView.RefreshInterval"] = md.MaterializedView.RefreshInterval.String()
 		metadata["MaterializedView.AllowNonIncrementalDefinition"] = strconv.FormatBool(md.MaterializedView.AllowNonIncrementalDefinition)
-		metadata["MaterializedView.MaxStaleness"] = md.MaxStaleness.String()
+		if md.MaxStaleness != nil {
+			metadata["MaterializedView.MaxStaleness"] = md.MaxStaleness.String()
+		}
+	}
+	if md.TimePartitioning != nil {
+		// "DAY", "HOUR", "MONTH", "YEAR"
+		metadata["TimePartitioning.Type"] = string(md.TimePartitioning.Type)
+		if md.TimePartitioning.Expiration != 0 {
+			metadata["TimePartitioning.Expiration"] = md.TimePartitioning.Expiration.String()
+		}
+		if md.TimePartitioning.Field != "" {
+			metadata["TimePartitioning.Field"] = md.TimePartitioning.Field
+		}
+	}
+	if md.RangePartitioning != nil {
+		if md.RangePartitioning.Field != "" {
+			metadata["RangePartitioning.Field"] = md.RangePartitioning.Field
+		}
+		if md.RangePartitioning.Range != nil {
```

Comment:
nit: in the previous block first check type and then field here first field and then range